File: Difficulty: Basic/LCM/lcm.java
// User function Template for Java
class Solution {
    public static int LCM(int a, int b) {
        int gcd = GCD(a, b);            // compute gcd
        return (a / gcd) * b;           // compute lcm using (a*b)/gcd, written like this to avoid overflow
    }

    // Helper method to compute GCD using Euclidean algorithm
    private static int GCD(int a, int b) {
        if (b == 0)
            return a;
        return GCD(b, a % b);
    }
}

File: Difficulty: Basic/LCM/lcm.cpp
// User function Template for C++

int LCM(int a, int b) {
   
    auto gcd = [](int x, int y) {
        while (y != 0) {
            int temp = y;
            y = x % y;
            x = temp;
        }
        return x;
    };
    
   
    return (a * b) / gcd(a, b);
}
